alt-tab-macos/logic/Preferences.swift

```diff
@@ -31,7 +32,7 @@ class Preferences {
     static var tabKeyCode: UInt16?
     static var highlightBorderColor: NSColor?
     static var highlightBackgroundColor: NSColor?
-    static var metaKeyCode: KeyCode?
+    static var metaKeyCodes: [UInt16] = []
```

Comment:
Could you replace this with `[UInt16]?`. I would like to avoid ambiguity of where things are initialized. I used to have all members of this class have some dummy value like this, but then it when changing the default values I would get confused on where the default values are.

Here we have a class that has no useful values until the JSON file is read or created on disk, so I think optional values represent that correctly. We know that after this file is read the optional are filled (otherwise the app is crashed and stopped), so we can use `!` everywhere in the app to query these preferences.

This is not ideal, but for some reason, Swift compiler doesn't notice that reading the JSON file happens in the initializer and that it has to happen or it throws. If the compiler was smarter, we could avoid the `?` since we know the object has to be initialized with actual values. But like I said between satisfying the compiler with random init values like `[]` or `.white` which could confuse the reader as default, and using `?` and later `!`, I would prefer the later approach

Reply:
Roger that. I was thinking about if it should be initialised or not but am not familiar enough with the Optional situation in Swift and chose the easy way out since there are other variables already initialised. Your point of view on that clears that up, thanks!